pkg/infrastructure/clusterapi/clusterapi.go

```diff
@@ -53,18 +53,28 @@ type Provider interface {
 	// such as IAM roles or policies.
 	PreProvision(in PreProvisionInput) error
 
+	// Ignition generates the ignition secrets for bootstrap and control-plane machines
+	// and handles any preconditions for ignition.
+	Ignition(in IgnitionInput) ([]client.Object, error)
+
 	// ControlPlaneAvailable is called once cluster.Spec.ControlPlaneEndpoint.IsValid()
 	// returns true, typically after load balancers have been provisioned. It can be used
 	// to create DNS records.
 	ControlPlaneAvailable(in ControlPlaneAvailableInput) error
 }
 
-// PreProvisionInput collects parameters to be passed to the PreProvision call.
+// PreProvisionInput collects the args to be passed to the PreProvision call.
 type PreProvisionInput struct {
 	ClusterID     string
 	InstallConfig *installconfig.InstallConfig
 }
 
+// IgnitionInput collects the args to be passed to the Ignition call.
+type IgnitionInput struct {
```

Comment:
Reviewing #7856, it looks like we would also need the installconfig here, which is fine. I will add it on my next go around.